### ISSUE TITLE
Changes to gateway name resolution break FQDN references

### ISSUE THREAD

**Issue by whitlaaa** [istio/istio#26264]
**Bug description**

I believe some changes to optimize `resolveGatewayName` as part of [0f49eca](https://github.com/istio/istio/commit/0f49eca87a408afb8e80f046f090546f412e4712#diff-e3109392080297ee093b7189648289e1L376) resulted in breaking changes when using FQDN for gateway references, causing routes to throw `NR` (No route configured) flags from Envoy until we switched to the `$NAMESPACE/$GATEWAY` format.

When upgrading our installation from 1.6.3 to 1.6.7 many of our routes broke that were using FQDNs and hadn't been updated to the `$NAMESPACE/$GATEWAY` format, yet. Specifically, it looks like the commit changed the resulting backwards compatibility output to be:

```
Gateway name:  private.ingress.svc.cluster.local
Before optimization:  ingress/private
After optimization:  ingress.svc.cluster.local/private
```

The above is output from a [Go Playground example](https://play.golang.org/p/AEFxDaKr_Yy) 

[ ] Docs
[ ] Installation
[ X ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

FQDN references for gateways continue to work for backwards compatibility

**Steps to reproduce the bug**

Upgrade to istio >= 1.6.6 with a VirtualService referencing a FQDN in its gateways. For example:

```yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  labels:
    env: prod
  name: my-service.example.com
  namespace: ingress
spec:
  gateways:
    - private.ingress.svc.cluster.local
    - mesh
 hosts:
    - my-service.example.com
 http:
   - route:
        - destination:
            host: private.ingress.svc.cluster.local
      timeout: 60s
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```shell
$ istioctl version --remote
client version: 1.6.7
control plane version: 1.6.7
data plane version: 1.6.7 (53 proxies)
```

```shell
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.6", GitCommit:"7015f71e75f670eb9e7ebd4b5749639d42e20079", GitTreeState:"clean", BuildDate:"2019-11-13T11:20:18Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:05:50Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```

**How was Istio installed?**

We use https://github.com/fluxcd/flux to manage the output of `istioctl manifest generate -f $CONFIG_FILE -o generated` , so effectively a `kubectl apply` of that output.

**Environment where bug was observed (cloud vendor, OS, etc)**

Azure cluster created via `aks-engine` directly (not AKS itself)

**Comment by howardjohn** [comment]
Thanks for the report! I opened https://github.com/istio/istio/pull/26329 to fix this, will backport to 1.6 as well

**Comment by whitlaaa** [comment]
@howardjohn Awesome, thanks for the quick response!
